source/extensions/tracers/xray/tracer.cc

```diff
@@ -0,0 +1,61 @@
+#include "extensions/tracers/xray/tracer.h"
+
+#include <algorithm>
+#include <array>
+#include <string>
+
+#include "envoy/http/header_map.h"
+
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+void Span::finishSpan() {
+  end_time_ = time_source_.systemTime();
+  // serialize this span and send it to X-Ray Daemon
```

Comment:
If this is for another PR, let's make it a todo comment in our standard format (`// TODO(marcomagdy): ...`)